client/components/table/index.js

```diff
@@ -2,164 +2,98 @@
 /**
  * External dependencies
  */
-import { __, sprintf } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
-import classnames from 'classnames';
-import { IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { IconButton, ToggleControl } from '@wordpress/components';
+import { noop } from 'lodash';
 import PropTypes from 'prop-types';
-import { isEqual, uniqueId } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
+import Card from 'components/card';
+import { EllipsisMenu, MenuItem, MenuTitle } from 'components/ellipsis-menu';
+import Pagination from 'components/pagination';
+import Table from './table';
+import TableSummary from './summary';
 
-const ASC = 'ascending';
-const DESC = 'descending';
+// @todo Handle toggling columns
 
-class Table extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			tabIndex: null,
-			rows: props.rows || [],
-			sortedBy: null,
-			sortDir: 'none',
-		};
-		this.container = createRef();
-		this.sortBy = this.sortBy.bind( this );
-		this.captionID = uniqueId( 'caption-' );
-	}
+const TableCard = ( {
+	headers,
+	onClickDownload,
+	onQueryChange,
+	query,
+	rows,
+	rowHeader,
+	summary,
+	title,
+} ) => {
+	return (
+		<Card
+			className="woocommerce-table"
+			title={ title }
+			action={
+				onClickDownload && (
+					<IconButton onClick={ onClickDownload } icon="arrow-down" size={ 18 } isDefault>
+						{ __( 'Download', 'wc-admin' ) }
+					</IconButton>
+				)
+			}
+			menu={
+				<EllipsisMenu label={ __( 'Choose which values to display', 'wc-admin' ) }>
+					<MenuTitle>{ __( 'Columns:', 'wc-admin' ) }</MenuTitle>
+					{ headers.map( ( label, i ) => (
+						<MenuItem key={ i } onInvoke={ noop }>
+							<ToggleControl label={ label } checked={ true } onChange={ noop } />
+						</MenuItem>
+					) ) }
+				</EllipsisMenu>
+			}
+		>
+			{ /* @todo Switch a placeholder view if we don't have rows */ }
```

Comment:
Since we have a default prop of `[]` for rows, will we also have an `isLoading` or similar prop to display a placeholder view properly? Thinking of scenarios where there is actually no data to show where we won't want to show a placeholder, but maybe some messaging around no results being found.

Reply:
The defaults are mostly so that it doesn't dump errors all over while loading, when we set up the loading state display I'd remove the `row` default.